art_internal_impl.hpp

```diff
@@ -485,11 +519,17 @@ union [[nodiscard]] key_prefix {
     UNODB_DETAIL_ASSERT(f.key_prefix_length.load() <= key_prefix_capacity);
   }
 
+  // Return the byte at the specified index.
   [[nodiscard]] constexpr auto byte_at(std::size_t i) const noexcept {
     UNODB_DETAIL_ASSERT(i < length());
     return f.key_prefix[i].load();
   }
 
+  // same as byte_at[].
+  [[nodiscard]] constexpr auto operator[](std::size_t i) const noexcept {
```

Comment:
My previous comment seems to be gone. Why we need both `byte_at` and `operator[]`? Can we just keep the latter?